base/sparse/sparsematrix.jl

```diff
@@ -2232,7 +2234,9 @@ getindex{T<:Integer}(A::SparseMatrixCSC, I::AbstractVector{Bool}, J::AbstractVec
 function setindex!{T,Ti}(A::SparseMatrixCSC{T,Ti}, v, i0::Integer, i1::Integer)
     i0 = convert(Ti, i0)
     i1 = convert(Ti, i1)
-    if !(1 <= i0 <= A.m && 1 <= i1 <= A.n); throw(BoundsError()); end
+    if !(1 <= i0 <= A.m && 1 <= i1 <= A.n)
+        throw(BoundsError())
```

Comment:
There's a `Base.throw_boundserror(A, inds)`. It's best to use this, because it has a `@noinline` in front of this, and this prevents the allocation of a GC frame whose only purpose is for error-reporting.